docs/rule_authors/writing_toolchains.md

```diff
@@ -0,0 +1,130 @@
+---
+id: writing_toolchains
+title: Writing Toolchains
+---
+
+Toolchains are regular rules that:
+
+- Have `is_toolchain_rule = True` passed to the `rule` call.
+- Return a toolchain provider, which is just a struct with a name like
+  `*ToolchainInfo`.
```

Comment:
What's special about these providers compared to others? I seem to be able to use `RunInfo` to define toolchains.

Reply:
I don't think there's anything special about them (someone may correct me though), the naming is just a convention.

A `RunInfo` would only work if you had a single command to run as part of your toolchain though, right? If that's the case, I guess it would work just fine, but this might be a bit confusing to rule authors, that would expect toolchains to be defined by providers that follow this convention?